base/random/generation.jl

```diff
@@ -301,13 +301,14 @@ rand!(rng::AbstractRNG, A::AbstractArray, r::UnitRange{<:Integer}) =
 
 ## random values from AbstractArray
 
-rand(rng::AbstractRNG, r::AbstractArray) = @inbounds return r[rand(rng, 1:length(r))]
 rand(                  r::AbstractArray) = rand(GLOBAL_RNG, r)
+rand(rng::AbstractRNG, r::AbstractArray) =
+    @inbounds return r[rand(rng, UnitRange(linearindices(r)))]
 
 ### arrays
 
 function rand!(rng::AbstractRNG, A::AbstractArray, r::AbstractArray)
-    g = RangeGenerator(1:(length(r)))
+    g = RangeGenerator(UnitRange(linearindices(r)))
```

Comment:
Any reason to force this to `UnitRange` rather than leaving as is? `RangeGenerator ` seems like it could take an `AbstractUnitRange`.

Reply:
I hesitated between the 2 options, and chose the laziest one! ok, will update `RangeGenerator` to accept `AbstractUnitRange`.